src/java/org/apache/cassandra/io/sstable/format/DataComponent.java

```diff
@@ -88,8 +88,12 @@ private static CompressionParams buildCompressionParams(TableMetadata metadata,
                 case fast:
                     if (!compressor.recommendedUses().contains(ICompressor.Uses.FAST_COMPRESSION))
                     {
-                        // The default compressor is generally fast (LZ4 with 16KiB block size)
-                        compressionParams = CompressionParams.DEFAULT;
+                        // The default compressor is generally Fast, but just in case we verify it.
+                        compressionParams = CompressionParams.defaultParams(metadata.keyspace);
+                        compressor = compressionParams.getSstableCompressor();
+                        if (!compressor.recommendedUses().contains(ICompressor.Uses.FAST_COMPRESSION)) {
+                            compressionParams = CompressionParams.FAST;
```

Comment:
might this change of the compressionParams surprise an operator ? 
(does it deserve a warning ? (one that would only be logged so often per table))